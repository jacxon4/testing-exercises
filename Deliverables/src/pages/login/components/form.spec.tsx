import * as React from 'react';
import { shallow } from 'enzyme';
import { FormProps } from './formProps';
import { createEmptyLoginFormErrors } from '../viewModel';
import { Form } from './form';

describe('pages/login/components/form', () => {
  it('should render properties provided', () => {
    const props: FormProps = {
      doLogin: () => {},
      loginEntity: {
        login: 'LoginEntityLogin',
        password: 'LoginEntityPassword',
      },
      loginFormErrors: createEmptyLoginFormErrors(),
      updateField: () => {},
    };

    const component = shallow(<Form {...props} />);

    expect(component).toMatchSnapshot();
  });
});
