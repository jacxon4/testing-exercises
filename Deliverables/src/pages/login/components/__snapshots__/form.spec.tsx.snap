// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/login/components/form should render properties provided 1`] = `
<form
  role="form"
>
  <fieldset>
    <Component
      error=""
      label="Login"
      name="login"
      onChange={[Function]}
      placeholder="E-mail"
      type="text"
      value="LoginEntityLogin"
    />
    <Component
      error=""
      label="Password"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value="LoginEntityPassword"
    />
    <Component
      label="Login"
      onClick={[Function]}
      type="submit"
    />
  </fieldset>
</form>
`;
